packages/core/lib/commands/debug/run.js

```diff
@@ -1,14 +1,19 @@
 module.exports = async function (options) {
   const { promisify } = require("util");
   const debugModule = require("debug");
+  const mergeConfigNetwork = require("./mergeConfigNetwork");
+  const loadConfig = require("./loadConfig");
   const debug = debugModule("lib:commands:debug");
 
   const { Environment } = require("@truffle/environment");
-  const Config = require("@truffle/config");
-
   const { CLIDebugger } = require("../../debug");
 
-  const config = Config.detect(options);
+  const config = loadConfig(options);
+
+  if (options.url) {
+    mergeConfigNetwork(config, options);
```

Comment:
Follow-on to previous comment, for consistency I'd suggest `config = mergeConfigNetwork(config, options);`.